core/src/replay_stage.rs

```diff
@@ -1467,20 +1467,30 @@ impl ReplayStage {
         drop(removed_banks);
 
         for (slot, slot_id) in slots_to_purge {
-            warn!(
-                "purging descendant: {} with slot_id {}, of slot {}",
-                slot, slot_id, duplicate_slot
-            );
             // Clear the slot signatures from status cache for this slot.
             // TODO: What about RPC queries that had already cloned the Bank for this slot
             // and are looking up the signature for this slot?
             root_bank.clear_slot_signatures(slot);
 
-            // Clear the slot-related data in blockstore. This will:
-            // 1) Clear old shreds allowing new ones to be inserted
-            // 2) Clear the "dead" flag allowing ReplayStage to start replaying
-            // this slot
-            blockstore.clear_unconfirmed_slot(slot);
+            if let Some(bank_hash) = blockstore.get_bank_hash(slot) {
```

Comment:
Maybe a comment on why we remove the ones with a bank hash from blockstore (because this version of the slot was successfully replayed  and chained from the duplicate so it must itself also be duplicate)